include/fty_common_rest_helpers.h

```diff
@@ -228,7 +262,20 @@ void check_user_permissions (
         http_errors_t &errors
         );
 
-#define CHECK_USER_PERMISSIONS_OR_DIE(p) \
+#define CHECK_USER_PERMISSIONS_OR_DIE (p) \
+    do { \
+        http_errors_t errors; \
+        std::string __http_die__debug__ {""}; \
+        if (::getenv ("BIOS_LOG_LEVEL") && !strcmp (::getenv ("BIOS_LOG_LEVEL"), "LOG_DEBUG")) { \
```

Comment:
Should there have been somewhere a macro/const/var for the strings in this macro, both the varname and the value  (to avoid spelling what is currently explicitly hardcoded)?

Also, I'd rather optimize this a bit (e.g. cache the return of `getenv` into a `char*` and null-compare/strcmp it) since the codepath is relatively hot (used at least once for every REST API call).

Same for the copy below.

Reply:
Possibly, yet again this should be addressed separately as is often the case. (this kind of change should be whole API related, not isolated to this file)